akka-typed/src/main/scala/akka/typed/persistence/scaladsl/PersistentActor.scala

```diff
@@ -41,19 +41,26 @@ object PersistentActor {
    * Factories for effects - how a persitent actor reacts on a command
    */
   object Effect {
-    def persist[Event, State](event: Event): Effect[Event, State] =
-      new Persist[Event, State](event)
 
-    def persistAll[Event, State](events: im.Seq[Event]): Effect[Event, State] =
+    def persist[Event, State](event: Event, events: Event*): Effect[Event, State] =
```

Comment:
Not sure that it matters in this context, but this will always allocate an empty array, for every call persisting a single element (which I think is the normal case). Would it be better to have a `persist(event)` and then `persist(event1, event2, events)` to avoid that?

Reply:
We can't have both: `persist(Event)` and `persist(Event*)` are ambiguous for the compiler

Reply:
We could have only `persist(evts: Event*)`, but then it will be legal to call `Effect.persist()`

Reply:
Further thought...

If we believe that a small optimisation like `Persist` and `PersistAll` should exist, then we should certainly avoid `persist(Event, Event*)`.

Therefore we should only offer `persist(Event)` and `persist(im.Seq[Event])`.

Reply:
That's the reason I said `persist(event1, event2, events)`, to use the arity to disambuigate the single parameter one. 

In the other API I think I did a `persistAll(evt1, evt2)` and `persistAll(evt1, evt2, evt3)` for convenience, if you have more events than that you probably already have a collection?

Reply:
Yeah I wonder, the slowest part here will be the persistence things in any case hm
Don't mind either way

Reply:
I think we should optimize for the common case, which is one single event.

Reply:
`persist(event1, event2, events)` ?

Reply:
Maybe I should have used a real signature instead, in case it's still unclear: `persist(evt1: Event, evt2: Event, events: Event*)` would cover `persistAll` as it would always be at least two events

Reply:
@johanandren I saw your comment on the other API. 
I will add the `persist(event1, event2, events)` variant.